caps/typeattr_path_test.go

```diff
@@ -0,0 +1,132 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"fmt"
+	"regexp"
+
+	. "gopkg.in/check.v1"
+)
+
+type pathSuite struct {
+	pathAttr    TypeAttr
+	devPathAttr TypeAttr
+	capType     *Type
+	cap         *Capability
+}
+
+var _ = Suite(&pathSuite{})
+
+func (s *pathSuite) SetUpSuite(c *C) {
+	// A path attribute that accepts any path
+	s.pathAttr = &pathAttr{
+		errorHint:       "path cannot be empty",
+		allowedPatterns: []*regexp.Regexp{regexp.MustCompile(".*")},
```

Comment:
I'm not sure I understand the pattern and the error hint. The pattern will work against the empty string too: https://play.golang.org/p/6QaYldG8aH - did you mean ".+" maybe? Or is this for something else?